internal/collector/config.go

```diff
@@ -102,3 +114,61 @@ func NewConfig(spec *v1beta1.InstrumentationSpec) *Config {
 
 	return config
 }
+
+// AddLogrotateConfig generates a logrotate configuration and adds it to the
+// provided configmap
+func AddLogrotateConfig(ctx context.Context, spec *v1beta1.InstrumentationSpec,
+	outInstanceConfigMap *corev1.ConfigMap, logFilePath, postrotateScript string,
+) error {
+	var err error
+	var retentionPeriod *v1beta1.Duration
+
+	if outInstanceConfigMap.Data == nil {
+		outInstanceConfigMap.Data = make(map[string]string)
+	}
+
+	// If retentionPeriod is set in the spec, use that value; otherwise, we want
+	// to use a reasonably short duration. Defaulting to 1 day.
+	if spec != nil && spec.Logs != nil && spec.Logs.RetentionPeriod != nil {
+		retentionPeriod = spec.Logs.RetentionPeriod
+	} else {
+		retentionPeriod, err = v1beta1.NewDuration("1d")
+		if err != nil {
+			return err
+		}
+	}
+
+	outInstanceConfigMap.Data["logrotate.conf"] = generateLogrotateConfig(logFilePath,
+		retentionPeriod, postrotateScript)
+
+	return err
+}
+
+// generateLogrotateConfig generates a configuration string for logrotate based
+// on the provided full log file path, retention period, and postrotate script
+func generateLogrotateConfig(logFilePath string, retentionPeriod *v1beta1.Duration,
+	postrotateScript string,
+) string {
+	number, interval := parseDurationForLogrotate(retentionPeriod)
+
+	return fmt.Sprintf(
+		logrotateConfigFormatString,
+		logFilePath,
+		number,
+		interval,
+		postrotateScript,
+	)
+}
+
+// parseDurationForLogrotate takes a retention period and returns the rotate
+// number and interval string that should be used in the logrotate config.
+// If the retentionPeriod is less than 24 hours, the function will return the
+// number of hours and "hourly"; otherwise, we will round up to the nearest day
+// and return the day count and "daily"
+func parseDurationForLogrotate(retentionPeriod *v1beta1.Duration) (int, string) {
+	hours := math.Round(retentionPeriod.AsDuration().Hours())
```

Comment:
🔧 We reject fractions of hours in the CRD right now, but this logic should still use ceiling here; e.g. `1.4h` → `2, "hourly"`

NB: 🤦🏻 There's also a bug in `strfmt` regarding fractions. If you add a test for this, use multiple units, like: `1 hour 20 min`.

```suggestion
	hours := math.Ceil(retentionPeriod.AsDuration().Hours())
```